composer/models/deeplabv3/deeplabv3.py

```diff
@@ -25,27 +42,53 @@ def deeplabv3_builder(num_classes: int,
         sync_bn (bool): if true (default), replace all BatchNorm layers with SyncBatchNorm layers.
     """
 
-    # Instantiate backbone module
-    if backbone_arch == 'resnet50':
-        resnet.model_urls[backbone_arch] = "https://download.pytorch.org/models/resnet50-f46c3f97.pth"
-    elif backbone_arch == 'resnet101':
-        resnet.model_urls[backbone_arch] = "https://download.pytorch.org/models/resnet101-cd907fc2.pth"
-    else:
-        raise ValueError(f"backbone_arch must be one of ['resnet50', 'resnet101'] not {backbone_arch}")
+    # check that the specified architecture is in the resnet module
+    if not hasattr(resnet, backbone_arch):
+        raise ValueError(f"backbone_arch must be part of the torchvision resnet module, got value: {backbone_arch}")
+
+    # change the model weight url if specified
+    if backbone_url:
+        resnet.model_urls[backbone_arch] = backbone_url
     backbone = getattr(resnet, backbone_arch)(pretrained=is_backbone_pretrained,
                                               replace_stride_with_dilation=[False, True, True])
-    backbone = _utils.IntermediateLayerGetter(backbone, return_layers={'layer4': 'out'})
 
-    # Instantiate head module
-    feat_extractor = ASPP(in_channels=2048, atrous_rates=[12, 24, 36], out_channels=256)
-    feat_extractor.project = feat_extractor.project[:3]  # Remove dropout due to higher standard deviation
-    classifier = torch.nn.Conv2d(in_channels=256, out_channels=num_classes, kernel_size=1)
-    head = torch.nn.Sequential(feat_extractor, classifier)
-
-    model = DeepLabV3(backbone, head, aux_classifier=None)
+    # specify which layers to extract activations from
+    return_layers = {'layer1': 'layer1', 'layer4': 'layer4'} if use_plus else {'layer4': 'layer4'}
+    backbone = _utils.IntermediateLayerGetter(backbone, return_layers=return_layers)
+
+    from mmseg.models import ASPPHead, DepthwiseSeparableASPPHead
+    norm_cfg = dict(type='SyncBN', requires_grad=True)
+    if use_plus:
+        # mmseg config:
+        # https://github.com/open-mmlab/mmsegmentation/blob/master/configs/_base_/models/deeplabv3plus_r50-d8.py
+        head = DepthwiseSeparableASPPHead(in_channels=2048,
+                                          in_index=-1,
+                                          channels=512,
+                                          dilations=(1, 12, 24, 36),
+                                          c1_in_channels=256,
+                                          c1_channels=48,
+                                          dropout_ratio=0.1,
+                                          num_classes=num_classes,
+                                          norm_cfg=norm_cfg,
+                                          align_corners=False)
+    else:
+        # mmseg config:
+        # https://github.com/open-mmlab/mmsegmentation/blob/master/configs/_base_/models/deeplabv3_r50-d8.py
+        head = ASPPHead(in_channels=2048,
```

Comment:
is this exactly the same as the torchvision `ASPP` head? or do we need to re-run our baselines.

Reply:
It isn't exactly the same. I think 2x the channels, dropout, and 1-2 more convolution layers. Rerunning baselines now